2021-04-28 11:16:31:97 :Helper : System.Exception: Getting Clients
2021-04-28 11:17:26:42 :Helper : System.Exception: Getting Clients
2021-04-28 11:18:13:97 :Logic Base : System.InvalidOperationException: The instance of entity type 'Client' cannot be tracked because another instance with the same key value for {'ClientID'} is already being tracked. When attaching existing entities, ensure that only one entity instance with a given key value is attached. Consider using 'DbContextOptionsBuilder.EnableSensitiveDataLogging' to see the conflicting key values.
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.ThrowIdentityConflict(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(TKey key, InternalEntityEntry entry, Boolean updateDuplicate)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(TKey key, InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IdentityMap`1.Add(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.StartTracking(InternalEntityEntry entry)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState oldState, EntityState newState, Boolean acceptChanges, Boolean modifyProperties)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState entityState, Boolean acceptChanges, Boolean modifyProperties, Nullable`1 forceStateWhenUnknownKey)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.PaintAction(EntityEntryGraphNode`1 node)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityEntryGraphIterator.TraverseGraph[TState](EntityEntryGraphNode`1 node, Func`2 handleNode)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityEntryGraphIterator.TraverseGraph[TState](EntityEntryGraphNode`1 node, Func`2 handleNode)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.AttachGraph(InternalEntityEntry rootEntry, EntityState targetState, EntityState storeGeneratedWithKeySetTargetState, Boolean forceStateWhenUnknownKey)
   at Microsoft.EntityFrameworkCore.DbContext.SetEntityState(InternalEntityEntry entry, EntityState entityState)
   at Microsoft.EntityFrameworkCore.DbContext.SetEntityState[TEntity](TEntity entity, EntityState entityState)
   at Microsoft.EntityFrameworkCore.DbContext.Add[TEntity](TEntity entity)
   at SmartLogic.TransactionService.CreatePayment(Transaction PaymentsFile, TransactionTypeList transaction) in D:\Dev Projects\SmartSave\SmartLogic\Service\TransactionService.cs:line 102
2021-04-28 11:28:18:41 :Helper : System.Exception: Getting Clients
2021-04-28 11:29:00:74 :Helper : System.Exception: Getting Clients
2021-04-28 11:40:20:99 :Helper : System.Exception: Getting Clients
2021-04-28 11:41:16:11 :Helper : System.Exception: Getting Clients
2021-04-28 11:56:25:88 :Helper : System.Exception: Getting Clients
2021-04-28 11:57:09:77 :Helper : System.Exception: Getting Clients
2021-04-28 12:57:37:43 :Helper : System.Exception: Getting Clients
2021-04-28 12:58:23:99 :Helper : System.Exception: Getting Clients
2021-04-28 13:04:37:98 :Helper : System.Exception: Getting Clients
2021-04-28 13:07:59:02 :Helper : System.Exception: Getting Clients
2021-04-28 13:13:44:04 :Helper : System.Exception: Getting Clients
2021-04-28 13:17:04:64 :Helper : System.Exception: Getting Clients
2021-04-28 13:19:39:72 :Helper : System.Exception: Getting Clients
2021-04-28 13:30:19:71 :Helper : System.Exception: Getting Clients
2021-04-28 14:58:58:58 :Helper : System.Exception: Getting Clients
2021-04-28 15:01:04:83 :Helper : System.Exception: Getting Clients
2021-04-28 15:01:41:68 :Helper : System.Exception: Getting Clients
2021-04-28 15:03:34:42 :Helper : System.Exception: Getting Clients
2021-04-28 15:09:14:77 :Helper : System.Exception: Getting Clients
2021-04-28 15:09:20:35 :Helper : System.Exception: Getting Clients
2021-04-28 15:13:42:76 :Helper : System.Exception: Getting Clients
2021-04-28 16:11:37:85 :Helper : System.Exception: Getting Clients
2021-04-28 16:11:56:41 :Helper : System.Exception: Getting Clients
2021-04-28 16:17:01:61 :Helper : System.Exception: Getting Clients
2021-04-28 16:22:01:67 :Helper : System.Exception: Getting Clients
2021-04-28 16:22:16:88 :Helper : System.Exception: Getting Clients
2021-04-28 16:29:52:05 :Helper : System.Exception: Getting Clients
2021-04-28 16:29:56:09 :Helper : System.Exception: Getting Clients
2021-04-28 16:31:13:45 :Helper : System.Exception: Getting Clients
2021-04-28 16:33:46:50 :Helper : System.Exception: Getting Clients
2021-04-28 16:42:36:17 :Helper : System.Exception: Getting Clients
2021-04-28 16:43:10:22 :Helper : System.Exception: Getting Clients
2021-04-28 16:51:27:36 :Helper : System.Exception: Getting Clients
2021-04-28 16:52:23:59 :Helper : System.Exception: Getting Clients
2021-04-28 16:53:43:99 :Helper : System.Exception: Getting Clients
2021-04-28 16:54:48:68 :Helper : System.Exception: Getting Clients
2021-04-28 17:56:36:57 :Helper : System.Exception: Getting Clients
2021-04-28 18:03:59:07 :Helper : System.Exception: Getting Clients
2021-04-28 18:05:31:67 :Helper : System.Exception: Getting Clients
2021-04-28 18:05:38:30 :Helper : System.Exception: Getting Clients
2021-04-28 20:07:45:87 :Helper : System.Exception: Getting Clients
2021-04-28 20:13:33:60 :Helper : System.Exception: Getting Clients
2021-04-28 20:14:10:01 :Helper : System.Exception: Getting Clients
2021-04-28 20:14:14:76 :Helper : System.Exception: Getting Clients
2021-04-28 20:14:33:31 :Helper : System.Exception: Getting Clients
2021-04-28 20:55:40:31 :Helper : System.Exception: Getting Clients
2021-04-28 20:56:18:03 :Helper : System.Exception: Getting Clients
2021-04-28 21:20:42:46 :Helper : System.Exception: Getting Clients
2021-04-28 21:21:18:43 :Helper : System.Exception: Getting Clients
2021-04-28 22:52:23:63 :Helper : System.Exception: Getting Clients
2021-04-28 22:52:30:14 :Logic Base : Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): The INSERT statement conflicted with the FOREIGN KEY constraint "FK_ClientDeductions_Products_ProductID". The conflict occurred in database "SmartSave", table "dbo.Products", column 'ProductID'.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__164_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__274_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:76a7413a-c68a-48af-b82b-9bc0197303cb
Error Number:547,State:0,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at SmartLogic.TransactionService.CreateInvoice(DateTime InvoiceDate, DateTime DueDate, Nullable`1 ProductID) in D:\Dev Projects\SmartSave\SmartLogic\Service\TransactionService.cs:line 389
2021-04-28 22:55:02:30 :Logic Base : Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details.
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): The INSERT statement conflicted with the FOREIGN KEY constraint "FK_ClientDeductions_Products_ProductID". The conflict occurred in database "SmartSave", table "dbo.Products", column 'ProductID'.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__164_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__274_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:76a7413a-c68a-48af-b82b-9bc0197303cb
Error Number:547,State:0,Class:16
   --- End of inner exception stack trace ---
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.ExecuteAsync(IEnumerable`1 commandBatches, IRelationalConnection connection, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(IList`1 entriesToSave, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChangesAsync(DbContext _, Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.ExecuteAsync[TState,TResult](TState state, Func`4 operation, Func`4 verifySucceeded, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChangesAsync(Boolean acceptAllChangesOnSuccess, CancellationToken cancellationToken)
   at SmartLogic.TransactionService.CreateInvoice(DateTime InvoiceDate, DateTime DueDate, Nullable`1 ProductID)
2021-04-28 22:57:35:12 :Helper : System.Exception: Getting Clients
2021-04-28 22:57:39:53 :Helper : System.Exception: Getting Clients
2021-04-28 22:57:53:60 :Helper : System.Exception: Getting Clients
2021-04-28 23:02:08:37 :Helper : System.Exception: Getting Clients
2021-04-28 23:02:18:42 :Helper : System.Exception: Getting Clients
2021-04-28 23:02:36:46 :Helper : System.Exception: Getting Clients
2021-04-28 23:07:57:13 :Helper : System.Exception: Getting Clients
2021-04-28 23:28:58:86 :Helper : System.Exception: Getting Clients
2021-04-28 23:29:03:10 :Helper : System.Exception: Getting Clients
2021-04-28 23:31:43:36 :Helper : System.Exception: Getting Clients
2021-04-28 23:33:55:49 :Helper : System.Exception: Getting Clients
